subsys/net/lib/dns/resolve.c

```diff
@@ -697,12 +733,30 @@ int dns_resolve_name(struct dns_resolve_context *ctx,
 		NET_DBG("DNS id will be %u", *dns_id);
 	}
 
+	/* If mDNS is enabled, then send .local queries only to multicast
+	 * address.
+	 */
+	if (IS_ENABLED(CONFIG_MDNS_RESOLVER)) {
+		const char *ptr = strstr(query, ".local");
 
-	for (j = 0; j < CONFIG_DNS_RESOLVER_MAX_SERVERS; j++) {
+		if (ptr && !memcmp(ptr, (const void *){ ".local\0" }, 7)) {
```

Comment:
Minimal libc has a `strrchr()` function: search for the last dot, and `memcmp()` the returned ptr with `".local"`. It should be faster than using `strstr()` which is quadratic in the minimal libc.